db/database.go

```diff
@@ -103,13 +104,19 @@ type OidcTestProviderOptions struct {
 type UserViewsOptions struct {
 	Enabled *bool `json:"enabled,omitempty"` // Whether pass-through view query is supported through public API
 }
+
 type UnsupportedOptions struct {
 	UserViews        UserViewsOptions        `json:"user_views,omitempty"`         // Config settings for user views
 	Replicator2      bool                    `json:"replicator_2,omitempty"`       // Enable new replicator (_blipsync)
 	OidcTestProvider OidcTestProviderOptions `json:"oidc_test_provider,omitempty"` // Config settings for OIDC Provider
 	EnableXattr      *bool                   `json:"enable_extended_attributes"`   // Use xattr for _sync
 }
 
+// Options associated with the import of documents not written by Sync Gateway
+type ImportOptions struct {
+	ImportFilter *ImportFilterFunction // Opt-in filter for document import
```

Comment:
It's a bit confusing to name a struct type with `Function` in the name.  I think maybe `ImportFilterJSServer` might be more descriptive and make it clear that there's some javascript behind this.